### PR TITLE
fix resource refs

### PR DESCRIPTION
**Pull request by avalanche123**
### Description of your changes

crossplane/crossplane#4858 changed crossplane behavior to generate resource
names purely client-side, which can lead to a resource ref not resolving for
a time. This PR handles not found error and skips pending resource refs from
responses.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR, as appropriate.~

### How has this code been tested

Going to test in a live cluster.

### CODE REVIEW

**Review by cbuto — APPROVED** [review]

File: schema/composite.gql [hunk 1]
```diff
@@ -150,6 +150,11 @@ type CompositeResourceSpec {
   """
   connectionSecret: Secret @goField(forceResolver: true)
 
+  """
+  The `ObjectReference`s for the resources composed by this composite resources.
+  """
+  resourceRefs: [ObjectReference!]!
```
Comment (jastang) [inline schema/composite.gql:156]: @thephred @miloszsobczak does the console need to use this new field?

File: internal/graph/resolvers/composite.go [hunk 2]
```diff
@@ -215,7 +215,9 @@ func (r *compositeResourceSpec) Resources(ctx context.Context, obj *model.Compos
 			xrc.SetKind(ref.Kind)
 			nn := types.NamespacedName{Namespace: ref.Namespace, Name: ref.Name}
 			if err := c.Get(ctx, nn, xrc); err != nil {
-				graphql.AddError(ctx, errors.Wrap(err, errGetComposed))
+				if !apierrors.IsNotFound(err) {
+					graphql.AddError(ctx, errors.Wrap(err, errGetComposed))
+				}
```
Comment (jastang) [inline internal/graph/resolvers/composite.go:220]: nit: add comment to xref the change in Crossplane's behaviour for context

**Review by jastang — APPROVED** [review]
thanks @avalanche123 !

File: schema/composite.gql (hunk 1 above)
Comment (avalanche123) [reply schema/composite.gql:156]: I added this for parity with other ref fields, but console could always use it using fieldPath or json fields

File: internal/graph/resolvers/composite.go [hunk 3]
```diff
@@ -215,7 +215,9 @@ func (r *compositeResourceSpec) Resources(ctx context.Context, obj *model.Compos
 			xrc.SetKind(ref.Kind)
 			nn := types.NamespacedName{Namespace: ref.Namespace, Name: ref.Name}
 			if err := c.Get(ctx, nn, xrc); err != nil {
-				graphql.AddError(ctx, errors.Wrap(err, errGetComposed))
+				if !apierrors.IsNotFound(err) {
```
Comment (thephred) [inline internal/graph/resolvers/composite.go:218]: An alternative would be to not consider not found a GraphQL error. We could either return a union that is the found object or not found maybe including the object ref or we could just not include not found refs as you have it here and rely on the consumer to compare with the objectrefs field to see what wasn't fetched.

Comment (avalanche123) [reply internal/graph/resolvers/composite.go:218]: for context, other refs are handled exactly this way:
* https://github.com/upbound/xgql/blob/main/internal/graph/resolvers/composite.go#L127-L129
* https://github.com/upbound/xgql/blob/main/internal/graph/resolvers/composite.go#L168C3-L170
* https://github.com/upbound/xgql/blob/main/internal/graph/resolvers/composite.go#L127-L129

**Comment by sttts** [comment]
Don't think https://github.com/crossplane/crossplane/pull/4858 changed anything here. It was server side before, but with dry-run. So it was a two-phase commit before as well, with the short chance to see dangling references. There are more such places in Crossplane with that pattern.

**Pull request merged**